pkg/prometheus/promcfg.go

```diff
@@ -2520,6 +2520,10 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	if sc.Spec.MetricRelabelConfigs != nil {
```

Comment:
I'd suggest this just to improve a bit readability.
```suggestion
	if sc.Spec.MetricRelabelConfigs != nil {
		relabelConfigValue = generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.MetricRelabelConfigs))
		cfg = append(cfg, yaml.MapItem{
			Key:   "metric_relabel_configs",
			Value: relabelConfigValue,
		})
	}	
```